File: JavaScript/01.JavaScript-Basic-February-2021/14.exams/2019.04.06_07-Exam/6.cinemaTickets.js
function cinemaTickets(input) {
    let studentTickets = 0;
    let standardTickets = 0;
    let kidsTickets = 0;
    let totalTickets = 0;

    for (let i = 0; i < input.length; i++) {
        if (input[i] == "End" || input[i] == "Finish") {
            break;
        }

        let movieTitle = input[i];
        i++;

        let freeSeats = Number(input[i]);
        i++;

        let currentMovieTickets = 0;
        while (input[i] != "End" && input[i] != "Finish") {            
            totalTickets += 1;
            currentMovieTickets += 1;
            switch (input[i]) {
                case "standard":
                    standardTickets += 1;
                    break;
                case "student":
                    studentTickets += 1;
                    break;
                case "kid":
                    kidsTickets += 1;
                    break;

                default:
                    break;
            }

            if (currentMovieTickets == freeSeats) {
                break;
            }

            i++;            
        }
        console.log(`${movieTitle} - ${(100 * currentMovieTickets / freeSeats).toFixed(2)}% full.`);
    }
    console.log(`Total tickets: ${totalTickets}`);
    console.log(`${(100 * studentTickets / totalTickets).toFixed(2)}% student tickets.`);
    console.log(`${(100 * standardTickets / totalTickets).toFixed(2)}% standard tickets.`);
    console.log(`${(100 * kidsTickets / totalTickets).toFixed(2)}% kids tickets.`);
}

cinemaTickets(["Taxi",
    "10",
    "standard",
    "kid",
    "student",
    "student",
    "standard",
    "standard",
    "End",
    "Scary Movie",
    "6",
    "student",
    "student",
    "student",
    "student",
    "student",
    "student",
    "Finish"]);


// cinemaTickets(["The Matrix",
//     "20",
//     "student",
//     "standard",
//     "kid",
//     "kid",
//     "student",
//     "student",
//     "standard",
//     "student",
//     "End",
//     "The Green Mile",
//     "17",
//     "student",
//     "standard",
//     "standard",
//     "student",
//     "standard",
//     "student",
//     "End",
//     "Amadeus",
//     "3",
//     "standard",
//     "standard",
//     "standard",
//     "Finish"]);

File: Python/01.Python-Basic-April-2019/15.exams/exam_2019.04.06/6.cinema_tickets.py
student = 0
standard = 0
kid = 0

film_name = input()
while film_name != 'Finish':
    place = int(input())

    people = 0
    for i in range(1, place + 1):
        tickets_type = input()
        if tickets_type == 'student':
            student += 1
        elif tickets_type == 'standard':
            standard += 1
        elif tickets_type == 'kid':
            kid += 1
        elif tickets_type == 'End':
            break
        people += 1

    full_place = people / place * 100
    print(f'{film_name} - {full_place:.2f}% full.')

    film_name = input()

total_tickets = student + standard + kid
student_tickets = (student / total_tickets) * 100
standard_tickets = (standard / total_tickets) * 100
kid_tickets = (kid / total_tickets) * 100

print(f'Total tickets: {total_tickets}')
print(f'{student_tickets:.2f}% student tickets.')
print(f'{standard_tickets:.2f}% standard tickets.')
print(f'{kid_tickets:.2f}% kids tickets.')
